dbms/src/DataStreams/SummingSortedBlockInputStream.cpp

```diff
@@ -156,7 +156,7 @@ SummingSortedBlockInputStream::SummingSortedBlockInputStream(
                 || endsWith(name, "Key")
                 || endsWith(name, "Type"))
             {
-                if (!nested_type.isValueRepresentedByInteger())
+                if (!nested_type.isValueRepresentedByInteger() && !isStringOrFixedString(nested_type))
```

Comment:
Ok.

Maybe we can remove this check at all?

How does it look for me:
- long time ago there was special code in SummingSortedBlockInputStream that was working only for integers;
- `sumMapWithOverflow` aggregate function was added;
- SummingSortedBlockInputStream was rewritten, to use `sumMapWithOverflow` in case of a single key;
- complex keys are implemented with `mergeMaps` method that is using FIelds and works for all types;

Do I understand it correctly?

Reply:
If some type is not supported, we will get exception from `sumMapWithOverflow` aggregate function and this is Ok.

Reply:
When you break here you get the behavior @den-crane described here: https://github.com/ClickHouse/ClickHouse/pull/8903#issuecomment-580531826 IMO this is not an optimal behavior, because it destroys data.

I tried removing that check altogether, and in that case if your key is say a tuple of strings, then you get an error from sumMap that the argument is illegal when you optimize the table... It feels a little late at that point to tell people we cannot take their data in.

Reply:
So where I'm getting at, is that I'm ok with removing that check, but that would be a breaking change. So I'll leave it up to you to decide. I'm fine with this PR as it is: it's introducing the least entropy as it's not disturbing any existing behavior, except for the string keys that it is now merging correctly.